### PR TITLE
Updated tutorial to latest versions of Syn components

### PR DESCRIPTION
**Pull request by akosma**
This PR updates the tutorial to the latest versions of Syn tools.

### CODE REVIEW

**Comment by akosma** [comment]
Ready for review!

**Review by ccremer — APPROVED** [review]
I didn't do a deep review here. 
I saw sometimes lots of "kubectl --context minikube" and I'm not sure if those commands still work with k8s-in-a-box providers other than minikube? Is the --context flag even required if we have a dedicated kubeconfig file?

**Comment by akosma** [comment]
> I didn't do a deep review here.
> I saw sometimes lots of "kubectl --context minikube" and I'm not sure if those commands still work with k8s-in-a-box providers other than minikube? Is the --context flag even required if we have a dedicated kubeconfig file?

Interesting, at least with minikube and k3d it seems to work fine. The tutorial isn't using a kubeconfig file, but we could indeed.

**Comment by ccremer** [comment]
> The tutorial isn't using a kubeconfig file, but we could indeed.

I would recommend doing that. Having a dedicated kubeconfig ensures no one needs to modify their existing ones and then throwing away the changes again after completing the tutorial.

File: docs/modules/ROOT/pages/index.adoc
Comment (simu) [inline docs/modules/ROOT/pages/index.adoc]:
```suggestion
In order to create a new component, we need to prepare a working directory for Commodore.
```

File: docs/modules/ROOT/pages/index.adoc
Comment (simu) [inline docs/modules/ROOT/pages/index.adoc]:
```suggestion
$ mkdir -p commodore
```

File: docs/modules/ROOT/pages/index.adoc
Comment (simu) [inline docs/modules/ROOT/pages/index.adoc]: This whole section seems unnecessary, we need something along the following lines instead:

```
==== Preparing the working directory

The easiest way to prepare a working directory for Commodore is to compile a cluster catalog.
This operation will create all the directories which Commodore expects to be present when creating a new component.
The following command will compile the catalog for one of the clusters which we've created previously.

[source,bash]
----
$ CLUSTER_ID=$(kubectl --context minikube -n lieutenant get cluster --no-headers -o custom-columns="NAME:.metadata.name" | head -n1)
$ commodore catalog compile ${CLUSTER_ID}
----

```

File: docs/modules/ROOT/pages/index.adoc
Comment (simu) [inline docs/modules/ROOT/pages/index.adoc]: Setting the owner here will correctly configure the remote of the new component's github repository to point to `git@github.com/${GITHUB_USERNAME}/component-fortune.git`

```suggestion
$ commodore component new fortune --owner ${GITHUB_USERNAME}
```

File: docs/modules/ROOT/pages/index.adoc
Comment (simu) [inline docs/modules/ROOT/pages/index.adoc:268]: Lines 291-292 (`git remote remove` and `git remote add`) are not required if we pass `--owner ${GITHUB_USERNAME}` to `component new`.

File: 1_lieutenant_on_minikube.sh
Comment (simu) [inline 1_lieutenant_on_minikube.sh:12]: Minikube chokes on Btrfs-on-LUKS because the Kubelet in the docker container cannot find the device for the cryptvolume in `/dev/mapper`. The error is something like:

```
Apr 26 15:50:53 minikube kubelet[2315]: W0426 15:50:53.878466    2315 fs.go:565] stat failed on /dev/mapper/nvme0n1p2_crypt with error: no such file or direct
ory
Apr 26 15:50:53 minikube kubelet[2315]: F0426 15:50:53.878479    2315 kubelet.go:1350] Failed to start ContainerManager failed to get rootfs info: failed to g
et device for dir "/var/lib/kubelet": could not find device with major: 0, minor: 29 in cached partitions map
```

Some googling found the following issue comment which suggests disabling a feature gate on sufficiently recent K8s versions: https://github.com/kubernetes/minikube/issues/7923#issuecomment-737058797'

This fixed the problem for me.

```suggestion
minikube start --disk-size 60g --cpus 4 --feature-gates="LocalStorageCapacityIsolation=false"
```

File: 4_synthesize_on_k3s.sh
Comment (simu) [inline 4_synthesize_on_k3s.sh:22]: Same story as for minikube, k3d doesn't like Btrfs-on-LUKS. Here the workaround is documented in https://github.com/k3s-io/k3s/issues/471#issuecomment-491810975

```suggestion
if [ -d /dev/mapper ]; then
k3d_extra_args="-v /dev/mapper:/dev/mapper"
else
k3d_extra_args=""
fi
k3d cluster create projectsyn ${k3d_extra_args}
```

**Review by simu — CHANGES_REQUESTED** [review]
LGTM overall.

However, when I gave it a test run locally I found that the setup scripts don't work out of the box on my machine (Btrfs-on-LUKS), see the inline comments on the respective shell scripts.

Possibly similar issues exist for the kind and microk8s scripts.

I also added some notes on a simplified way to setup the Commodore working directory and the component repo's git remote for component development.

File: docs/modules/ROOT/pages/index.adoc
Comment (akosma) [reply docs/modules/ROOT/pages/index.adoc]: Oh I thought that the commodore project had to be checked out, just a folder is enough then? Cool!

File: docs/modules/ROOT/pages/index.adoc
Comment (akosma) [reply docs/modules/ROOT/pages/index.adoc:268]: Excellent, thanks!

File: 4_synthesize_on_k3s.sh
Comment (akosma) [reply 4_synthesize_on_k3s.sh:22]: Is it common to use Btrfs-on-LUKS?

File: docs/modules/ROOT/pages/index.adoc
Comment (simu) [reply docs/modules/ROOT/pages/index.adoc]: Yeah, since we're using the Commodore docker image in the tutorial, the project doesn't need to be checked out.

File: 4_synthesize_on_k3s.sh
Comment (simu) [reply 4_synthesize_on_k3s.sh:22]: No idea, I'm also not sure if that's the only setup where missing devices in `/dev/mapper` become an issue, reading the comments it seems that some LVM setups experience similar issues.

**Comment by akosma** [comment]
> > The tutorial isn't using a kubeconfig file, but we could indeed.
> 
> I would recommend doing that. Having a dedicated kubeconfig ensures no one needs to modify their existing ones and then throwing away the changes again after completing the tutorial.

At least Minikube and K3d both overwrite the local ~/.kube/config upon start, and they remove their respective entries when finished. I suggest using this approach for the moment and update later with a kubeconfig file strategy in a separate PR.

File: 4_synthesize_on_k3s.sh
Comment (akosma) [reply 4_synthesize_on_k3s.sh:22]: If you're ok with this, I'd rather add both this and the previous minikube information as part of the README, for "Troubleshooting" purposes, in case somebody experiments the same, would that be ok?

**Comment by akosma** [comment]
Having closed all conversations, waiting for @tobru 's approval and we can merge this.

**Review by tobru — APPROVED** [review]

**Pull request merged**
